set-client/src/main/java/eu/europeana/set/client/connection/BaseApiConnection.java

```diff
@@ -1,156 +1,340 @@
 package eu.europeana.set.client.connection;
 
-import java.io.IOException;
-
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import eu.europeana.api.commons.definitions.search.result.impl.ResultsPageImpl;
+import eu.europeana.set.client.exception.SetApiClientException;
+import eu.europeana.set.client.model.result.AbstractUserSetApiResponse;
+import eu.europeana.set.client.model.result.RecordPreview;
+import eu.europeana.set.common.http.HttpConnection;
+import eu.europeana.set.definitions.model.UserSet;
+import eu.europeana.set.definitions.model.impl.BaseUserSet;
+import eu.europeana.set.definitions.model.vocabulary.ProfileConstants;
+import eu.europeana.set.definitions.model.vocabulary.WebUserSetFields;
+import jakarta.ws.rs.core.UriBuilder;
```

Comment:
should be replaced by the use of rg.apache.hc.core5.net.URIBuilder

Reply:
done